vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -656,26 +714,75 @@ void AdsImpl::OnAdNotificationTimedOut(const AdNotificationInfo& ad) {
       {ad.segment, AdNotificationEventType::kTimedOut});
 }
 
+void AdsImpl::OnAdNotificationEventFailed(
+    const std::string& uuid,
+    const AdNotificationEventType event_type) {
+  BLOG(1, "Failed to fire ad notification " << event_type << " event for uuid "
+                                            << uuid);
+}
+
 void AdsImpl::OnNewTabPageAdViewed(const NewTabPageAdInfo& ad) {
   account_->Deposit(ad.creative_instance_id, ConfirmationType::kViewed);
 }
 
 void AdsImpl::OnNewTabPageAdClicked(const NewTabPageAdInfo& ad) {
-  ad_transfer_->set_last_clicked_ad(ad);
+  ad_transfer_->SetLastClickedAd(ad);
 
   account_->Deposit(ad.creative_instance_id, ConfirmationType::kClicked);
 }
 
+void AdsImpl::OnNewTabPageAdEventFailed(
+    const std::string& uuid,
+    const std::string& creative_instance_id,
+    const NewTabPageAdEventType event_type) {
+  BLOG(1, "Failed to fire new tab page ad "
+              << event_type << " event for uuid " << uuid
+              << " and creative instance id " << creative_instance_id);
+}
+
 void AdsImpl::OnPromotedContentAdViewed(const PromotedContentAdInfo& ad) {
   account_->Deposit(ad.creative_instance_id, ConfirmationType::kViewed);
 }
 
 void AdsImpl::OnPromotedContentAdClicked(const PromotedContentAdInfo& ad) {
-  ad_transfer_->set_last_clicked_ad(ad);
+  ad_transfer_->SetLastClickedAd(ad);
+
+  account_->Deposit(ad.creative_instance_id, ConfirmationType::kClicked);
+}
+
+void AdsImpl::OnPromotedContentAdEventFailed(
+    const std::string& uuid,
+    const std::string& creative_instance_id,
+    const PromotedContentAdEventType event_type) {
+  BLOG(1, "Failed to fire promoted content ad "
+              << event_type << " event for uuid " << uuid
+              << " and creative instance id " << creative_instance_id);
+}
+
+void AdsImpl::OnDidServeInlineContentAd(const InlineContentAdInfo& ad) {
+  inline_content_ad_->FireEvent(ad.uuid, ad.creative_instance_id,
+                                InlineContentAdEventType::kServed);
+}
+
+void AdsImpl::OnInlineContentAdViewed(const InlineContentAdInfo& ad) {
+  account_->Deposit(ad.creative_instance_id, ConfirmationType::kViewed);
+}
+
+void AdsImpl::OnInlineContentAdClicked(const InlineContentAdInfo& ad) {
+  ad_transfer_->SetLastClickedAd(ad);
 
   account_->Deposit(ad.creative_instance_id, ConfirmationType::kClicked);
```

Comment:
same as above

Reply:
These do not need to be guarded as promoted and inline content ads are part of Brave News Ads and send private confirmations if ads is disabled or regular confirmations if ads is enabled